hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -25,30 +296,418 @@ The following table lists the main parameters used throughout the OTP controller
+The OTP storage in `prim_generic_otp` is emulated using a standard RAM primitive `prim_generic_ram_1p`.
+While this storage element is volatile, the primitive is constructed such that the contents are not wiped upon a system-wide reset.
+I.e., only a power-cycle wipes the RAM primitive, thereby enabling limited emulation of the OTP function and life cycle transitions also on an FPGA device.
+
+
+# Programmer's Guide
+
+During provisioning and manufacturing, SW interacts with the OTP controller mostly through the Direct Access Programming interface, which is described below.
+Afterwards during production, SW is expected to perform only read accesses via the exposed CSRs and CSR windows, since all write access to the partitions has been locked down.
+
+The following sections provide some general guidance, followed by an explanation of the direct access interface and a detailed OTP memory map.
+Typical programming sequences are explained at the end of the Programmer's guide.
 
+## General Guidance
 
-## Power-up and Reset Considerations
+### Initialization
 
-## Initialization
+The OTP controller initializes automatically upon power-up and is fully operational by the time the processor boots.
+The only initialization steps that SW should perform are:
 
-## Interrupt Handling
+1. Check that the OTP controller has successfully initialized by making sure that {{< regref STATUS >}} is IDLE and not in an ERROR state.
+2. Program and lock down the frequency of periodic background checks in HW.
 
+Step 2. can be achieved by adjusting the MSB indices for the pseudo-randomly drawn periods via {{< regref INTEGRITY_CHECK_PERIOD_MSB >}} and {{< regref CONSISTENCY_CHECK_PERIOD_MSB >}}.
+It is recommended to leave this configuration at the default setting, but SW must under all circumstances write to the {{< regref CHECK_PERIOD_REGEN >}} register in order to lock write access to these CSRs.
+
+Later on during the boot process, SW may also choose to block read access to the {{< regref CREATOR_SW_CFG >}} or {{< regref OWNER_SW_CFG >}} partitions at runtime via {{< regref CREATOR_SW_CFG_READ_LOCK >}} and {{< regref OWNER_SW_CFG_READ_LOCK >}}.
+
+### Reset Considerations
+
+It is important to note that values in OTP **can be corrupted** if a reset occurs during a programming operation.
+This should be of minor concern for SW, however, since all partitions except for the LIFE_CYCLE partition are being provisioned in secure and controlled environments, and not in the field.
+The LIFE_CYCLE partition is the only partition that is modified in the field - but that partition is entirely owned by the life cycle controller and not by SW.
+
+### Programming Already Programmed Regions
+
+OTP words cannot be programmed twice, and doing so may damage the memory array.
+Hence the OTP controller performs a blank check and returns an error if a write operation is issued to an already programmed location.
```

Comment:
Is this error the "Programming error occurred" error you mentioned at line 450?

Reply:
Yes, I this is one of the error conditions that could lead to 0x7. Depending on how many error conditions we can have during the programming step, we may have to split these up into multiple codes...